File: parent-337/pinyougou-interface/src/main/java/cn/ithcast/core/service/GoodsService.java
package cn.ithcast.core.service;

import cn.ithcast.core.pojo.good.Goods;
import cn.ithcast.core.pojo.seckill.SeckillGoods;
import cn.ithcast.core.pojogroup.GoodsVo;
import entity.PageResult;
import entity.Result;

public interface GoodsService {

    /**
     * 添加商品
     * @param goodsVo
     */
    void add(GoodsVo goodsVo);

    /**
     * 分页+条件查询商品
     * @param pageNum
     * @param pageSize
     * @param goods
     * @return
     */
    PageResult search(Integer pageNum, Integer pageSize, Goods goods);

    /**
     * 根据商品id查询
     * @param id
     * @return
     */
    GoodsVo findOne(Long id);

    /**
     * 修改商品
     * @param goodsVo
     */
    void update(GoodsVo goodsVo);

    /**
     * 运营商批量修改商品状态
     * @param selectIds
     * @param id
     */
    void updateStatus(Long[] selectIds, String id);

    /**
     * 运营商批量删除商品
     * @param selectIds
     */
    void del(Long[] selectIds);


    Result addSeconds(Long[] ids,SeckillGoods seckillGoods);
}

File: parent-337/pinyougou-web-manager/src/main/webapp/js/service/goodsService.js
app.service('goodsService', function ($http) {
    this.add = function (entity) {
        return $http.post('../goods/add.do', entity);
    };

    this.update = function (entity) {
        return $http.post('../goods/update.do', entity);
    };
    // 分页+条件查询
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../goods/search.do?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity);
    };

    // 根据商品id查询
    this.findOne = function (id) {
        return $http.get('../goods/findOne.do?id=' + id);
    };

    // 批量修改商品状态
    this.updateStatus = function (selectIds, id) {
        return $http.get('../goods/updateStatus.do?selectIds=' + selectIds + '&id=' + id);
    };

    // 批量删除商品
    this.del = function (selectIds) {
        return $http.get('../goods/del.do?selectIds=' + selectIds);
    }
});